### ISSUE TITLE
[4][Keegan-lee] Customizable Table Component

### ISSUE THREAD

**Issue by sschiessl-bcp** [bitshares/bitshares-ui#2761]
As seen from discussion here https://github.com/bitshares/bitshares-ui/issues/2655#issuecomment-499369911

If @Keegan-lee takes this, the task is clear as already discussed.

**Comment by Keegan-lee** [comment]
Yes, I would like this issue.

**Comment by startailcoon** [comment]
> 
> 
> Yes, I would like this issue.

Assigned

**Comment by startailcoon** [comment]
I'd like to point out that I'm still not sold on the concept of solving the space issue with making it possible to view/hide different actions. An action that I seldom use would be much more convinient to have in a drop-down action so I can still access them, and still save space. After this change I have to enable it based on my need for the moment.

But if this is what we want to go, I'm not going to hinder it, just want to give my view on this.

![](https://user-images.githubusercontent.com/13334179/59794563-a1067d80-92af-11e9-9eaa-2aaaf7359982.png)

**Comment by sschiessl-bcp** [comment]
I agree @startailcoon , but the customization possibility will be helpful either way. We will ahve to re-iterate on the UX when completed.

**Comment by startailcoon** [comment]
Noticed something when trying this today @Keegan-lee @sschiessl-bcp 

When you scroll down in the list to toggle something, the list will scroll back to the top again. This is annoying if you're removing all the actions at the end of the list. 

Also, can we get rid of the double header? When you open the drop down it has the title again as first option in the list. Why?

While you're at the spot, can you please rename "Percentage of Total Supply" to something short? Use a tooltip for description if needed.

**Comment by sschiessl-bcp** [comment]
@Keegan-lee please check and fix if needed. Add hours if needed as well. You can rename to "% of Total Supply"

**Comment by startailcoon** [comment]
> % of Total Supply"

Can't we omit "Total" and just have `% of Supply`?

**Comment by Keegan-lee** [comment]
@startailcoon - It has the first title as the first option because essentially, that is the ONLY option that can ever be selected. The dropdown menu never changes, and just provides a way to toggle the table columns.

@sschiessl-bcp - I will fix the scroll jumping.

**Comment by startailcoon** [comment]
By the double header I mean that the drop down say "Customize the columns" but the first option say the same thing and is an option that can't be selected.

![image](https://user-images.githubusercontent.com/12114550/62487709-cbaa8800-b7c2-11e9-8a68-a00b3b4736cb.png)

This should be possible to be without, example on https://ant.design/components/select/
![image](https://user-images.githubusercontent.com/12114550/62488021-8dfa2f00-b7c3-11e9-8d89-d6459a02076a.png)

----

Then we need to fix how "Search" and "Active"/"Visual" behaves on a smaller screen as the new drop down will cover these when the screen turns smaller. This would be a good new issue, so we can move this to the next issue linked to this one instead of adding more work to this.

I would suggest 
- Search gets a small search icon and expands on request for input. 

![image](https://user-images.githubusercontent.com/12114550/62487074-2c38c580-b7c1-11e9-8785-b81f55c8075b.png)

**Comment by sschiessl-bcp** [comment]
How about

![image](https://user-images.githubusercontent.com/33128181/62768250-bf5e4d80-ba96-11e9-9c18-9276c68231b8.png)
![image](https://user-images.githubusercontent.com/33128181/62768295-d735d180-ba96-11e9-9ef9-6d5e350b9367.png)
